crates/pink/sidevm/host-runtime/src/run.rs

```diff
@@ -79,16 +79,22 @@ impl Future for WasmRun {
     type Output = Result<i32, RuntimeError>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-        self.env.set_gas_to_breath(self.gas_per_breath);
-        match async_context::set_task_cx(cx, || self.wasm_poll_entry.call()) {
-            Ok(rv) => {
-                if rv == 0 {
-                    Poll::Pending
-                } else {
-                    Poll::Ready(Ok(rv))
+        loop {
+            self.env.set_gas_to_breath(self.gas_per_breath);
+            match async_context::set_task_cx(cx, || self.wasm_poll_entry.call()) {
```

Comment:
Does it mean we are going to poll until the current task has finished all its tasks, or exhausted the gas? In the past we only poll once and move to the next task.

Reply:
Yes, at the moment.
The loop was in the guest side past. Moving it out to here so that we can let the sidevm scheduler handle it in the future.

Reply:
However, it should still work to remove the loop in the current code base. Just need to add a `waker.wake()`.

Reply:
Removed the loop.